api/src/room_store.py

```diff
@@ -1,29 +1,69 @@
 import os
 import json
+from typing import Protocol
+from abc import abstractmethod
 
 
-class RoomStore:
-    def __init__(self, room_store_dir):
-        self.path = room_store_dir
+class RoomStore(Protocol):
+    path: str
+
+    @abstractmethod
+    def get_all_room_ids(self) -> list:
+        raise NotImplementedError
+
+    @abstractmethod
+    def write_room_data(self, room_id: str, data: dict) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def read_room_data(self, room_id: str) -> dict:
+        raise NotImplementedError
+
+    @abstractmethod
+    def room_data_exists(self, room_id: str) -> bool:
+        raise NotImplementedError
```

Comment:
The examples I saw just put ... for the bodies and didn't have the annotations, Does intellij complain or something if you do that?

Reply:
According to PEP 544 (https://www.python.org/dev/peps/pep-0544/#protocol-members), "An abstract method that should not be called via super() ought to raise NotImplementedError"